--- icu_host/source/i18n/double-conversion-utils.h	2019-04-17 12:42:15.000000000 -0700
+++ icu/source/i18n/double-conversion-utils.h	2019-04-19 21:36:29.757338554 -0700
@@ -103,7 +103,7 @@
     defined(__AARCH64EL__) || defined(__aarch64__) || defined(__AARCH64EB__) || \
     defined(__riscv) || \
     defined(__or1k__) || defined(__arc__) || \
-    defined(__EMSCRIPTEN__)
+    defined(__EMSCRIPTEN__) || defined(__wasm__)
 #define DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS 1
 #elif defined(__mc68000__) || \
     defined(__pnacl__) || defined(__native_client__)
